base/abstractarray.jl

```diff
@@ -2403,7 +2403,14 @@ function hash(A::AbstractArray, h::UInt)
     # Instead hash the tuple of firsts and lasts along each dimension
     h = hash(map(first, axes(A)), h)
     h = hash(map(last, axes(A)), h)
-    isempty(A) && return h
+
+    # For short arrays, it's not worth doing anything complicated 
+    if length(A) < 10000
+        @inbounds for p in pairs(IndexLinear(),A)
+            h ⊻= hash(p)
```

Comment:
Although if we don't include the key, this will probably need to be `h = hash(p, h)` to prevent evenly repeated values from hashing the same.

Reply:
Yeah. The reason I wanted to do it this way was so the `hash` computations would be independent. This theoretically should allow the compiler to re-order this to interweave hash computation.

Reply:
Is it cheaper to do half the hashes but use the `h = hash(x, h)` form? We could/should remove the key hashing below, too.

Reply:
Good call. Updated